src/front/app/components/Challenges/ChallengeList.js

```diff
@@ -1,7 +1,49 @@
 import React from 'react';
+import { connect } from 'react-redux';
+import actions from './../../actions/challenges';
+import Challenge from './Challenge';
+import TopChallenge from './TopChallenge';
 
-export default () => (
-    <div>
-        ChallengeList
-    </div>
-);
+class ChallengeList extends React.Component {
+    constructor(props) {
+        super(props);
+        const { challenges, dispatch } = props;
+        if (!challenges.length) {
```

Comment:
Please look at AsyncApp component example http://redux.js.org/docs/advanced/ExampleRedditAPI.html for  better async data flow handling in component (note the required props)